src/query-parser.ts

```diff
@@ -1,18 +1,31 @@
 import { Dictionary } from "@navarik/types"
 import { SearchQuery } from "./types"
 
+const castType = (value, type) => {
+  if (!type){return value}
+  if (type.includes('int') || type.includes('long')) {
+    return Number(value)
+  }
+  else if (type.includes('boolean')) {
+    return Boolean(value)
+  }
+  else return value
+}
+
 export class QueryParser {
-  parse(searchParams: Dictionary<any>): SearchQuery {
+  parse(searchParams: Dictionary<any>, schemas: object): SearchQuery {
+
     if (searchParams.operator && searchParams.args) {
       return <SearchQuery>searchParams
     }
 
     const args: Array<any> = []
     for (const field in searchParams) {
+      const typeLookUp = field =="type" ? null: schemas[field]
       const value = searchParams[field]
       const term = value.operator
-        ? value
-        : { operator: "literal", args: [value] }
+        ? castType(value, typeLookUp)
```

Comment:
const value = castType(searchParams[field], typeLookUp)
const term = value.operator
        ? value
        : { operator: "literal", args: [value] }

Reply:
Should have caught that. lazy brain